docs/node/production-guide.mdx

```diff
@@ -0,0 +1,392 @@
+
+Alternatively, check the new node's health check REST endpoint, which will return an HTTP status code of 200 and a JSON body with `"healthy":true` if the node is ready:
+
+```sh title="node2"
+[node2]:~$ curl -s --fail http://127.0.0.1:8484/api/v1/health
+{"kwil_alive":true,"healthy":true, ...}
+```
+
+#### Joining the Validator Set
+
+Unlike `node0` and `node1`, consensus does not require participation from `node2`, which just executes the blocks that are produced and approved by the validators. In this section we will add `node2` to the validator set.
+
+Start by submitting a validator "join" request from `node2`:
+
+```sh title="node2"
+[node2]:~$ kwil-admin validators join 
+TxHash: 0722630c4296ea27d77f4ee8db6129c1ca042372811187f7ba787c75464400ed
+```
+
+We have now submitted a request to become a validator, which the existing validators must approve for this node to join the validator set. The `validators list-join-requests` subcommmand from any node can view this request:
+
+```sh
+$ kwil-admin validators list-join-requests
+Pending join requests (1 approval needed):
+ Candidate                                                        | Power | Approvals | Expiration
+------------------------------------------------------------------+-------+-----------+------------
+ 7c02692ed4e1f165791b43742274edb4f9587f6da02d54a0c111386b7110bb64 |     1 |         0 | 14627
+```
+
+Both validators are required to approve this request since 1/2 does not meet the threshold of greater than 2/3 of the validators. This is done by running the following on **each validator**:
+
+```sh
+$ kwil-admin validators approve 7c02692ed4e1f165791b43742274edb4f9587f6da02d54a0c111386b7110bb64
+TxHash: cab4b66d12cd6ecee256ba73dd9500a3ea7b21ba03bd38d5db7701268a0f8ecb
+```
+
+Once the approval transactions are mined, `node2` will become part of the validator set:
+
+```sh
+$ kwil-admin validators list
+Current validator set:
+  0. {pubkey = 8e11018570091d79b6105d5954cbbc4f1570cd3a02df2214b362e7c3f22c0459, power = 1}
+  1. {pubkey = 1e3ab63c62ad165d01627d53c6c4c61e59626ceeeb26653527cf3df18eff7a23, power = 1}
+  2. {pubkey = 7c02692ed4e1f165791b43742274edb4f9587f6da02d54a0c111386b7110bb64, power = 1}
+```
+
+On the block where the final approval transaction was mined, the `kwild` logs for `node2` will display:
+
+```plain
+{"level":"info","ts":1727997606.6760225,"logger":"kwild.cometbft","msg":"finalized block","module":"state","height":291,"num_txs_res":1,"num_val_updates":1,"block_app_hash":"F67E090714D70095DB4DF0EF3BF592A8E60FEC0CEEE58B5FB4D55CBD92E8BC40"}
+{"level":"info","ts":1727997606.6823015,"logger":"kwild.cometbft","msg":"updates to validators","module":"state","updates":"2EA4B9CA4A6FCBAF5102E4C8AB335D785F5D4772:1"}
+{"level":"info","ts":1727997606.6865957,"logger":"kwild.listener-manager","msg":"Node is a validator and caught up with the network, starting listeners"}
+```
+
+The node is now a validator.  Keep it running and monitor it's health.
```

Comment:
I'm realizing that this document entirely neglects to mention version considerations.  From current Slack discussions right now, I think it is worth mentioning something along the lines of :

> Between "major" kwil releases, the application's state machine logic generally changes, so this is accompanied by a bump to the "app protocol version", which is that 0 -> 1  change you see in the block header.
> During the developement cycle (changes on main branch between releases) there are continual changes, so in a production environment where the network is not disposable, only release versions (or release-v0.x branches) would be run.  The protocol version is there to signal when there is different application logic (e.g. execution rules) that make the different versions incompatible on the same network.
> But if you had a network that had been running say v0.8 for months and wanted to upgrade to v0.9, this would involve a migration.

Reply:
Do you think this is needed in the production deployment guide? Or in an upgrading guide?